### PR TITLE
enabling tidyOutput by default

### PR DESCRIPTION
**Pull request by fredyns**
i'd like to set tidy-output by default to make the generated code more readable.

### CODE REVIEW

**Comment by schmunk42** [comment]
I need to think about enabling `tidyOutput` by default. Might be a good option.
But the function `getCoreProviders` should return **core** providers only.

Can you see why tests are working only with the second commit?

**Comment by fredyns** [comment]
thanks for your consideration.

i'll do something about `getCoreProviders` function.
maybe i can duplicate it or make it execute per folder.

sorry, i'm not used with Travis-CI & getpocket.com.
(mentioned in the log)
i'll try to figure it out.

**Comment by fredyns** [comment]
does this line cause the test not working?
it was `array`

```
/**
   * @var bool whether to use phptidy on renderer files before saving
   */
```

File: src/generators/crud/default/views/view.php [hunk 1]
```diff
@@ -10,16 +10,17 @@
 
 /** @var \yii\db\ActiveRecord $model */
 /** @var $generator \schmunk42\giiant\generators\crud\Generator */
-
 ## TODO: move to generator (?); cleanup
-$model = new $generator->modelClass();
+$model          = new $generator->modelClass();
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Could you reformat the indents for variables assignment, will also point to some other...

File: src/generators/crud/default/views/view.php [hunk 2]
```diff
@@ -134,27 +141,29 @@
...
         $showAllRecords = false;
 
-        if ($relation->via !== null) {
-            $pivotName = Inflector::pluralize($generator->getModelByTableName($relation->via->from[0]));
+        if ($relation->via !== null)
+        {
+            $pivotName     = Inflector::pluralize($generator->getModelByTableName($relation->via->from[0]));
             $pivotRelation = $model->{'get'.$pivotName}();
-            $pivotPk = key($pivotRelation->link);
+            $pivotPk       = key($pivotRelation->link);
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Indent

File: src/generators/crud/default/views/view.php [hunk 3]
```diff
@@ -165,49 +174,53 @@
...
+            $pjaxId       = "pjax-{$pivotName}";
             $gridRelation = $pivotRelation;
-            $gridName = $pivotName;
-        } else {
-            $pjaxId = "pjax-{$name}";
+            $gridName     = $pivotName;
+        }
+        else
+        {
+            $pjaxId       = "pjax-{$name}";
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Indent

File: src/generators/crud/default/views/view.php [hunk 4]
```diff
@@ -165,49 +174,53 @@
...
             $gridRelation = $relation;
-            $gridName = $name;
+            $gridName     = $name;
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Indent

File: src/generators/crud/providers/core/RelationProvider.php [hunk 5]
```diff
@@ -324,20 +324,26 @@ public function relationGrid($name, $relation, $showAllRecords = false)
         $pageParam = Inflector::slug("page-{$name}");
         $firstPageLabel = $this->generator->generateString('First');
         $lastPageLabel = $this->generator->generateString('Last');
-        $code = '\'<div class="table-responsive">\' . ';
+        $code           = "'<div class=\"table-responsive\">'\n . ";
```
Comment (schmunk42) [inline src/generators/crud/providers/core/RelationProvider.php]: Indent

**Comment by fredyns** [comment]
done fixing those indent.
now i'll do something about the failed test.

**Comment by schmunk42** [comment]
Tests are passing... Looks good to me. Any more to do from your side?

**Comment by fredyns** [comment]
oh... great :D
nope, the code is neat.
thank you.

File: src/generators/crud/Generator.php [hunk 6]
```diff
@@ -393,7 +393,7 @@ public function render($template, $params = [])
             $tmpFile = $tmpDir.'/'.md5($template);
             file_put_contents($tmpFile, $code);
 
-            shell_exec('vendor'.DIRECTORY_SEPARATOR.'bin'.DIRECTORY_SEPARATOR.'phptidy replace '.$tmpFile);
+            shell_exec('..'.DIRECTORY_SEPARATOR.'vendor'.DIRECTORY_SEPARATOR.'bin'.DIRECTORY_SEPARATOR.'phptidy replace '.$tmpFile);
```
Comment (schmunk42) [inline src/generators/crud/Generator.php]: On a second look, maybe `Yii::getAlias` would be the right thing here.

File: src/generators/crud/default/views/index.php [hunk 7]
```diff
@@ -95,8 +93,8 @@
     <?= "<?php \yii\widgets\Pjax::begin(['id'=>'pjax-main', 'enableReplaceState'=> false, 'linkSelector'=>'#pjax-main ul.pagination a, th a', 'clientOptions' => ['pjax:success'=>'function(){alert(\"yo\")}']]) ?>\n"; ?>
 
     <h1>
-        <?= "<?= Yii::t('{$generator->modelMessageCategory}', '{$modelName}') ?>" ?>
-        <small>
+            <?= "<?= Yii::t('{$generator->modelMessageCategory}', '{$modelName}') ?>\n" ?>
+            <small>
             List
```
Comment (schmunk42) [inline src/generators/crud/default/views/index.php]: The original looks more correct to me, any special reason for this indent?

File: src/generators/crud/default/views/index.php [hunk 8]
```diff
@@ -188,16 +187,15 @@
...
         <?php if ($generator->searchModelClass !== ''): ?>
             'filterModel' => $searchModel,
         <?php endif; ?>
         'tableOptions' => ['class' => 'table table-striped table-bordered table-hover'],
         'headerRowOptions' => ['class'=>'x'],
-        'columns' => [
-
-        <?php
+                'columns' => [
+            <?php
```
Comment (schmunk42) [inline src/generators/crud/default/views/index.php]: Same here...

File: src/generators/crud/default/views/view.php [hunk 9]
```diff
@@ -10,16 +10,17 @@
...
-if (empty($safeAttributes)) {
+if (empty($safeAttributes))
+{
     $model->setScenario('default');
     $safeAttributes = $model->safeAttributes();
 }
-if (empty($safeAttributes)) {
+if (empty($safeAttributes))
+{
     $safeAttributes = $model->getTableSchema()->columnNames;
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: `if` code should be PSR-2 compatible, see https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md

File: src/generators/crud/default/views/view.php [hunk 10]
```diff
@@ -61,9 +62,9 @@
 
 
     <h1>
-        <?= "<?= Yii::t('{$generator->modelMessageCategory}', '{$modelName}') ?>" ?>
+        <?= "<?= Yii::t('{$generator->modelMessageCategory}', '{$modelName}') ?>\n" ?>
         <small>
-            <?= '<?= $model->'.$generator->getModelNameAttribute($generator->modelClass).' ?>' ?>
+                        <?= '<?= $model->'.$generator->getModelNameAttribute($generator->modelClass)." ?>\n" ?>
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Also here, original looks more correct.

File: src/generators/crud/default/views/view.php [hunk 11]
```diff
@@ -98,20 +99,24 @@
     <hr />
 
     <?php
-    echo "<?php \$this->beginBlock('{$generator->modelClass}'); ?>\n";
+    echo "<?php \$this->beginBlock('".substr(strrchr($model::className(), '\\'), 1)."'); ?>\n";
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Do we really need to sanitize the model-name in the block?

File: src/generators/crud/default/views/view.php [hunk 12]
```diff
@@ -134,24 +141,26 @@
...
     // get relation info $ prepare add button
     $model = new $generator->modelClass();
-
-    $items = <<<EOS
+    $label = substr(strrchr($model::className(), '\\'), 1);
+$items = <<<EOS
 [
     'label'   => '<b class=""># '.\$model->{$model->primaryKey()[0]}.'</b>',
-    'content' => \$this->blocks['{$generator->modelClass}'],
+    'content' => \$this->blocks['{$label}'],
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Sanitation not required, like above.

File: src/generators/crud/default/views/view.php [hunk 13]
```diff
@@ -134,24 +141,26 @@
...
-    foreach ($generator->getModelRelations($generator->modelClass, ['has_many']) as $name => $relation) {
+    foreach ($generator->getModelRelations($generator->modelClass, ['has_many']) as $name => $relation)
+    {
         echo "\n<?php \$this->beginBlock('$name'); ?>\n";
 
         $showAllRecords = false;
 
-        if ($relation->via !== null) {
+        if ($relation->via !== null)
+        {
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: PSR-2

File: src/generators/crud/default/views/view.php (hunk 11 above)
Comment (fredyns) [inline src/generators/crud/default/views/view.php]: the rest block is well sanitized.
i think it's good to apply the same pattern.

**Comment by fredyns** [comment]
a had look all modification thouroughly.
fixing all indentation.

File: src/generators/crud/default/views/view.php (hunk 11 above)
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: But this is just an internal array key for preparing content via blocks, there's no output of this value. And you loose information, could be a problem to have two models with the same name in different namespaces.

File: src/generators/crud/default/views/view.php [hunk 14]
```diff
@@ -137,13 +137,14 @@
 
     // get relation info $ prepare add button
     $model = new $generator->modelClass();
-
+    $label = substr(strrchr($model::className(), '\\'), 1);
```
Comment (schmunk42) [inline src/generators/crud/default/views/view.php]: Same sanitation as above

**Comment by schmunk42** [comment]
Great work, thank you very much!

**Comment by fredyns** [comment]
your welcome.

btw, this is great extension.
hope i could contribute more.

**Comment by schmunk42** [comment]
If you could give the local dockerized testing a try, this would be nice. Just to see if it works for you.

**Pull request merged**